dbms/src/Storages/DeltaMerge/DMContext.h

```diff
@@ -103,9 +162,9 @@ struct DMContext : private boost::noncopyable
         bool is_common_handle_,
         size_t rowkey_column_size_,
         const DB::Settings & settings,
-        const ScanContextPtr scan_context_ = nullptr,
+        const ScanContextPtr & scan_context_ = nullptr,
         const String & tracing_id_ = "")
-        : db_context(db_context_)
+        : global_context(session_context_.getGlobalContext()) // always save the global context
```

Comment:
always save the global context no matter what the pass-in `session_context` is a session only context or global context, and rename it to be `global_context`